src/sdam/topology_description.ts

```diff
@@ -431,7 +431,7 @@ function updateRsWithPrimaryFromMember(
   setName?: string
 ): TopologyType {
   if (setName == null) {
-    throw new MongoDriverError('setName is required');
+    throw new MongoInvalidArgumentError('setName is required');
```

Comment:
Weirdly this seems to me to be a MongoCompatibilityError, we've been using that to mean version incompatibility but here it would mean topology incompatibility. I'm fine with leaving this as is, but its an interesting case of if the setName is missing then we're not connected to a replica set so trying to `updateRsWithPrimaryFromMember` is illogical. 

But anyway this is internal logic that ideally would be caught in testing, it wouldn't surface